### PR TITLE
Unify parallel task launches

### PR DESCRIPTION
**Pull request by phillipm**
Replace all instances of 
```
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
    mTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR, textMessageBody);
} else {
    mTask.execute(textMessageBody);
}
```
with
```
mTask.executeParallel(textMessageBody);
```

And make some tasks that were launched using `execute` be launched using the parallel invocation `executeParallel`

### CODE REVIEW

File: app/src/org/commcare/engine/resource/AndroidResourceManager.java [hunk 1]
```diff
@@ -261,7 +261,7 @@ private void launchRetryTask(Context ctx) {
                 UpdateTask updateTask = UpdateTask.getNewInstance();
                 updateTask.startPinnedNotification(ctx);
                 updateTask.setAsAutoUpdate();
-                updateTask.execute(ref);
+                updateTask.executeParallel(ref);
```
Comment (phillipm) [inline app/src/org/commcare/engine/resource/AndroidResourceManager.java]: Need to add this change to 2.27 LTS

File: app/src/org/commcare/activities/EntitySelectActivity.java [hunk 2]
```diff
@@ -482,7 +482,7 @@ public boolean loadEntities() {
         if (loader == null && !EntityLoaderTask.attachToActivity(this)) {
             EntityLoaderTask entityLoader = new EntityLoaderTask(shortSelect, asw.getEvaluationContext());
             entityLoader.attachListener(this);
-            entityLoader.execute(selectDatum.getNodeset());
+            entityLoader.executeParallel(selectDatum.getNodeset());
```
Comment (phillipm) [inline app/src/org/commcare/activities/EntitySelectActivity.java]: Need to add this change to 2.27 LTS

Comment (amstone326) [inline app/src/org/commcare/activities/EntitySelectActivity.java]: has this happened yet?

Comment (phillipm) [inline app/src/org/commcare/activities/EntitySelectActivity.java]: Decided we wanted to do QA before releasing this

**Pull request merged**